File: Problem/Programmers/DFS_Q1.py
answer = 0


def DFS(idx, numbers, target, curSum):
    global answer

    if idx == len(numbers) and target == curSum:
        answer += 1
        return
    if idx == len(numbers):
        return

    DFS(idx + 1, numbers, target, curSum + numbers[idx])
    DFS(idx + 1, numbers, target, curSum - numbers[idx])


def solution(numbers, target):
    global answer

    DFS(0, numbers, target, 0)
    return answer

File: Problem/Programmers/dfs_bfs/Dfs_Q1.java

// 타겟 넘버
class Solution {
    int answer = 0;
    
    public void dfs(int[] numbers, int target, int curSum, int curIdx){
        if(curIdx == numbers.length){
            if(target == curSum){
                answer++;
            }
        }else{
            dfs(numbers,target,curSum+numbers[curIdx],curIdx+1);
            dfs(numbers,target,curSum-numbers[curIdx],curIdx+1);
        }
    }
    public int solution(int[] numbers, int target) {
        
        dfs(numbers, target, 0, 0);
        return answer;
    }
}